docs/types.rst

```diff
@@ -751,13 +751,12 @@ Members
 ^^^^^^^
 
 **length**:
-    Arrays have a ``length`` member to hold their number of elements.
-    Dynamic arrays can be resized in storage (not in memory) by changing the
-    ``.length`` member. This does not happen automatically when attempting to access elements outside the current length. The size of memory arrays is fixed (but dynamic, i.e. it can depend on runtime parameters) once they are created.
+    Fixed-length arrays (storage or memory) have a read-only ``length`` member to access their number of elements.
+    Dynamically-sized arrays (only available for storage) have a read-write ``length`` member to resize the array. Increasing the length adds uninitialized elements to the array, this has O(1) complexity. Reducing the length performs ``delete`` on each removed element and is O(n) complexity where n is the number of elements being deleted. Please note that calling ``length--`` on an empty array will set the length of the array to 2^256-1 due to uint256 underflow wrapping.
```

Comment:
Syntactical improvement:
- please put `O(1)`, `O(n)` and `n` into emphasis (eg. `*O(1)`)
- and `uint256` into backticks